server/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java

```diff
@@ -909,7 +909,11 @@ public IndexFieldData.Builder fielddataBuilder(FieldDataContext fieldDataContext
             } else if (operation == FielddataOperation.SEARCH) {
                 if (fielddata == false) {
                     throw new IllegalArgumentException(
-                        "Text fields are not optimised for operations that require per-document "
+                        "Field data is disabled on ["
+                            + name()
+                            + "] in ["
+                            + fieldDataContext.fullyQualifiedIndexName()
+                            + "]. Text fields are not optimised for operations that require per-document "
                             + "field data like aggregations and sorting, so these operations are disabled by default. Please use a "
```

Comment:
@javanna Is this `field data` ok to have the white space between the words? If so, I will leave this as it is. Otherwise, I will also change here and the corresponding part in the test.